test/tokengenerator.go

```diff
@@ -0,0 +1,141 @@
+package testutils
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/lestrrat-go/jwx/jwk"
+)
+
+const (
+	bitSize = 2048
+)
+
+// WebKeySet represents a JWK Set object.
+type WebKeySet struct {
+	Keys []jwk.Key `json:"keys"`
+}
+
+// ExtraClaim a function to set claims in the token to generate
+type ExtraClaim func(token *jwt.Token)
+
+// WithEmailClaim sets the `email` claim in the token to generate
+func WithEmailClaim(email string) ExtraClaim {
+	return func(token *jwt.Token) {
+		token.Claims.(jwt.MapClaims)["email"] = email
+	}
+}
+
+// Identity is a user identity
+type Identity struct {
+	ID       uuid.UUID
+	Username string
+}
+
+// NewIdentity returns a new, random identity
+func NewIdentity() *Identity {
+	return &Identity{
+		ID:       uuid.NewV4(),
+		Username: "testuser-" + uuid.NewV4().String(),
+	}
+}
+
+// TokenGenerator represents the test token and key manager.
+type TokenGenerator struct {
+	keyMap map[string]*rsa.PrivateKey
+}
+
+// NewTokenGenerator creates a new TokenGenerator.
+func NewTokenGenerator() *TokenGenerator {
+	tg := &TokenGenerator{}
+	tg.keyMap = make(map[string]*rsa.PrivateKey)
+	return tg
+}
+
+// CreateKey creates and stores a new key with the given kid.
+func (tg *TokenGenerator) CreateKey(kid string) (*rsa.PrivateKey, error) {
```

Comment:
you could rename it as `AddPrivateKey`, so we know it not only generates, but also stores it

Reply:
fixed here: 99f9f5a29e9fc882d73d514867a92a72773f5289